odbc/src/api/functions.rs

```diff
@@ -380,16 +389,53 @@ pub extern "C" fn SQLDriverConnect(
 
 #[no_mangle]
 pub extern "C" fn SQLDriverConnectW(
-    _connection_handle: HDbc,
+    connection_handle: HDbc,
     _window_handle: HWnd,
-    _in_connection_string: *const WChar,
-    _string_length_1: SmallInt,
+    in_connection_string: *const WChar,
+    string_length_1: SmallInt,
     _out_connection_string: *mut WChar,
     _buffer_length: SmallInt,
     _string_length_2: *mut SmallInt,
     _driver_completion: DriverConnectOption,
 ) -> SqlReturn {
-    unimplemented!()
+    return SqlReturn::SUCCESS;
+    println!("______");
+    let conn_handle = MongoHandleRef::from(connection_handle);
+    let conn = (*conn_handle).as_connection();
+    if conn.is_none() {
+        conn_handle.add_diag_info(ODBCError::InvalidHandleType("Connection"));
+        return SqlReturn::ERROR;
```

Comment:
According to the [ODBC Docs](https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqldriverconnect-function?view=sql-server-ver16) it appears that both `SQLDriverConnect` and `SQLDisconnect` have more error cases than the ones we've implemented here

Let's make sure we cover the stuff that's been documented and also unit/integration test them as needed to get coverage!

Reply:
Definitely. I want to make sure to handle the MoreInfo cases, too.

Reply:
Filled SQL-991 to handle this so that we can start parallelizing the other work!